cypress/e2e/bookStore.cy.js

```diff
@@ -1,11 +1,66 @@
 /// <reference types='cypress' />
 
 describe('Book Store app', () => {
+  const testData = {
+    user: {
+      username: 'Testyty',
+      password: 'M113m8J1p4z$'
+    },
+    book: {
+      name: 'Speaking JavaScript',
+      author: 'Axel Rauschmayer',
+      publisher:'O\'Reilly Media',
+      description: 'Like it or not, JavaScript is everywhere these days-from browser to server to mobile-and now you'
+    },
+  };
+
   before(() => {
-    
+    cy.visit('https://demoqa.com/login');
   });
 
-  it('', () => {
-    
+  it('login', () => {
+    cy.get('[placeholder="UserName"]')
+      .type(testData.user.username);
+    cy.get('[placeholder="Password"]')
+      .type(testData.user.password)
+    cy.contains('.btn-primary', 'Login')
+      .click();
+    cy.get('#userName-value')
+      .should('contain', testData.user.username)
+    cy.url()
+      .should('contain', '/profile');
+    });
+
+    it('Should add book', () => {
+      cy.login(testData.user.username, testData.user.password);
+      cy.contains('#item-2', 'Book Store')
+        .click();
+      cy.get('[placeholder="Type to search"]')
+        .type(testData.book.name);
+      cy.contains('[role="row"]', testData.book.name)
+        .should('contain', testData.book.author)
+        .and('contain', testData.book.publisher);
+      cy.contains('a', testData.book.name)
+        .click();
+      cy.get('#description-wrapper')
+        .should('contain', testData.book.description);
+      cy.contains('#addNewRecordButton', 'Add To Your Collection')
+        .click();
+      cy.on('window:alert', (alert) => {
+        expect(alert).to.eq('Book added to your collection.');
+      });
+    });
+    it('should delete a book',() => {
+      cy.login(testData.user.username, testData.user.password);
+      cy.contains('#item-3', 'Profile')
+        .click();
+      cy.contains('[role="row"]', testData.book.name)
+        .find('[title="Delete"]')
+        .click();
+      cy.get('closeSmallModal-ok')
```

Comment:
There is a typo in the selector, it should be '#closeSmallModal-ok' instead of 'closeSmallModal-ok'.